### PR TITLE
Olivia Charles- PR 

### PR DESCRIPTION
**Pull request by ochar721**

### CODE REVIEW

File: lib/shopping_cart.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  end
+
+  def products_by_category(category)
+    @products.find_all do |product|
+      product.category == category
+    end
+  end
+
+  def percentage_occupied
+   ((total_number_of_products/ @capacity.to_f) * 100).round(1)
```
Comment (mikedao) [inline lib/shopping_cart.rb:44]: We wanted this out to the hundredth place but we'll let you have it.

File: lib/shopping_cart.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
+class ShoppingCart
+  attr_reader :name,
+              :capacity,
+              :products
+
+  def initialize(name, capacity)
+    @name     = name
+    @capacity = capacity.delete("items").to_i
```
Comment (mikedao) [inline lib/shopping_cart.rb:8]: Great

File: lib/shopping_cart.rb [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  end
+
+  def add_product(item_name)
+    @products << item_name
+  end
+
+  def details
+    {
+    name: @name,
+    capacity: @capacity
```
Comment (mikedao) [inline lib/shopping_cart.rb:19]: Clean

File: lib/shopping_cart.rb [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+      product.quantity
+    end
+  end
+
+  def is_full?
+    if total_number_of_products > @capacity
+      true
+    else
+      false
+    end
```
Comment (mikedao) [inline lib/shopping_cart.rb:34]: Can be shortened to total_number_of_products > @capacity

**Review by mikedao — COMMENTED** [review]
Great work.

Score: 4

File: lib/shopping_cart.rb (hunk 1 above)
Comment (ochar721) [reply lib/shopping_cart.rb:44]: thanks I read that wrong :|

**Pull request closed without merging**